cirq/sim/act_on_state_vector_args.py

```diff
@@ -227,3 +228,35 @@ def _strat_act_on_state_vector_from_mixture(action: Any,
                                   out=args.available_buffer)
     args.swap_target_tensor_for(args.available_buffer)
     return True
+
+
+def _strat_act_on_state_vector_from_channel(action: Any,
+                                            args: 'cirq.ActOnStateVectorArgs'
+                                           ) -> bool:
+    kraus_operators = protocols.channel(action, default=None)
+    if kraus_operators is None:
+        return NotImplemented
+
+    shape = protocols.qid_shape(action)
+    kraus_tensors = [
+        e.reshape(shape * 2).astype(args.target_tensor.dtype)
+        for e in kraus_operators
+    ]
+    p = args.prng.random()
+    weight = None
+    for i in range(len(kraus_tensors)):
```

Comment:
is there a version of this in which `i` is recorded?

Reply:
Not at the moment.